src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -447,9 +502,16 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             Arguments.ManifestResources,
                             emitOptions,
                             debugEntryPoint: null,
+                            testData: null,
                             getHostDiagnostics: getAnalyzerDiagnostics,
                             cancellationToken: cancellationToken);
 
+                        if (!emitResult.Success && (newCompilation != null) && (newCompilation != compilation))
+                        {
+                            compilation = newCompilation;
+                            goto retry;
```

Comment:
The `retry` label is above at line 321. The `goto` could be replaced by a `while` loop but using a `goto` avoided changing the indenting of a significant amount of code, making the change simpler to read. Added a comment explaining the retry.